README.md

```diff
@@ -118,6 +118,65 @@ chunks they are interested in.  The columns chunks should then be read sequentia
 
  ![File Layout](https://raw.github.com/apache/parquet-format/master/doc/images/FileLayout.gif)
 
+ ### PAR3 File Footers
+
+ PAR3 file footer footer format designed to better support wider-schemas and more control
+ over the various footer size vs compute trade-offs.  Its format is as follows:
+   - Data pages containing serialized Thrift metadata objects that were modeled as lists
+     in PAR1.These are stored contiguously with offsets stored in the FileMetadata. See 
+     parquet.thrift for more details on each.
+   - Serialized Thrift FileMetadata Structure
+   - (Optional) 4 byte CRC32 of the serialized Thrift FileMetadata.
+   - 4-byte length in bytes (little endian) of the serialized FileMetadata structure.
+   - 4-byte length in bytes (little endian) of all preceding elements in the footer.
+   - 1 byte flag field to indicate features that require special parsing of the footer.
+     Readers MUST raise an error if there is an unrecognized flag.  Current flags:
+
+     * 0x01 - Footer encryption enabled (when set the encryption information is written before 
+        FileMeta structure as in the PAR1 footer).
+     * 0x02 - CRC32 of FileMetadata Footer.
+
+   - 4-byte magic number "PAR3"
+
+  When parsing the footer implementations SHOULD read at least the last 10 bytes of the footer. Then
+  read in the entirety of the footer based on the length of all preceding elements. This prevents further
+  I/O cost for accessing metadata stored in the data pages. PAR3 footers can fully replace PAR1 footers.
+  If a file is written with only PAR3 footer, implementation MUT write PAR3 as the first four bytes in 
+  they file. PAR3 footers can also be written in a backwards compatible way after PAR1 Metadata 
+  (see next section for details).
+
+  #### Dual Mode PAR1 and PAR3 footers
+
+  There is a desire to gradually rollout PAR3 footers to allow newer readers to take advantage of them, while
+  older readers can still properly parse the file. This section outlines a strategy to do this.
+
+  As backgroud, Thrift structs are always serialized with a 0 trailing byte do delimit there ending.  
```

Comment:
Does it mean a 0 bytes cannot appear anywhere else in a Thrift serialization? Is this mentioned somewhere in the Thrift specs?

Reply:
Othere fields a length delimited.  Docs: https://github.com/apache/thrift/blob/master/doc/specs/thrift-compact-protocol.md#struct-encoding

Reply:
Sorry I think I might have misunderstood your question.  I thought you were asking if 0x00 can appear elsewhere in serialized data, and the answer is yes, but it doesn't introduce ambiguity.

Reply:
But then, what is the point of mentioning a trailing byte above?

Reply:
Sorry I tried to clean up some grammar/typos and this might have gone out of date.  

I'm not sure what reference you are referring to (I'll reread and try to do another pass).  I think the reasons for mentioning it are either:
1.  A prior iteration of this which described a way to gaurantee a thrift-parser would ignore the PAR3 footer by simulating an unknown field ID.
2. As an explanation to build up-to why we believe the current proposal is safe.  In short: FileMetadata is a struct.  When serialized it will always end with 0x00.  As long as implementations do not do any padding with PAR3 after serializaing FileMetadata today, "PAR3" after the offset the offset is an unambiguous signal that a PAR3 footer is present.

Reply:
I know this is a draft PR, but I would find this easier to read if the README concentrated on describing the format at a high level view, rather than explaining specifics of Thrift encoding. If need be, those can be detailed in a separate design document (perhaps a RFC-like document? see also the typical structure of [Python PEPs](https://peps.python.org/pep-0000/)).

Reply:
Ok, I'll rephrase

Reply:
simplified to just the layout with a TODO to decide on final layout based on other proposal.